python/pyarrow/tests/test_dataset.py

```diff
@@ -391,6 +393,26 @@ def test_dataset(dataset, dataset_reader):
     assert sorted(result['group']) == [1, 2]
     assert sorted(result['key']) == ['xxx', 'yyy']
 
+    condition = ds.field(('struct', 'b')) == '1'
+    result = dataset.to_table(use_threads=True, filter=condition).to_pydict()
+
+    assert result['i64'] == [1, 4, 1, 4]
```

Comment:
This one could be in another order, you should probably sort the result table before (using `Table.sort_by` perhaps).

Reply:
Sure, will do 👍

Reply:
Ok, I pushed a commit to show better what I meant :-)

Reply:
🤦‍♀️🤷‍♀️ Thank you!